pkg/integrations/v2/app_o11y_receiver/sourcemaps/sourcemaps.go

```diff
@@ -0,0 +1,331 @@
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("unexpected status %v", resp.StatusCode)
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return body, nil
+}
+
+func (store *RealSourceMapStore) downloadSourceMapContent(sourceURL string) (content []byte, resolvedSourceMapURL string, err error) {
+	level.Debug(store.l).Log("msg", "attempting to download source file", "url", sourceURL)
+
+	result, err := store.downloadFileContents(sourceURL)
+	if err != nil {
+		level.Debug(store.l).Log("failed to download source file", "url", sourceURL, "err", err)
+		return nil, "", err
+	}
+	r := regexp.MustCompile(reSourceMap)
+	match := r.FindAllStringSubmatch(string(result), -1)
+	if len(match) == 0 {
+		level.Debug(store.l).Log("msg", "no sourcemap url found in source", "url", sourceURL)
+		return nil, "", nil
+	}
+	sourceMapURL := match[len(match)-1][2]
+
+	// inline sourcemap
+	if strings.HasPrefix(sourceMapURL, "data:") {
+		dataURL, err := dataurl.DecodeString(sourceMapURL)
+		if err != nil {
+			level.Debug(store.l).Log("msg", "failed to parse inline sourcemap data url", "url", sourceURL, "err", err)
+			return nil, "", err
+		}
+
+		level.Info(store.l).Log("msg", "successfully parsed inline sourcemap data url", "url", sourceURL)
+		return dataURL.Data, sourceURL + ".map", nil
+	}
+	// remote sourcemap
+	resolvedSourceMapURL = sourceMapURL
+
+	// if url is relative, attempt to resolve absolute
+	if !strings.HasPrefix(resolvedSourceMapURL, "http") {
+		base, err := url.Parse(sourceURL)
+		if err != nil {
+			level.Debug(store.l).Log("msg", "failed to parse source url", "url", sourceURL, "err", err)
+			return nil, "", err
+		}
+		relative, err := url.Parse(sourceMapURL)
+		if err != nil {
+			level.Debug(store.l).Log("msg", "failed to parse source map url", "url", sourceURL, "sourceMapURL", sourceMapURL, "err", err)
+			return nil, "", err
+		}
+		resolvedSourceMapURL = base.ResolveReference(relative).String()
+		level.Debug(store.l).Log("msg", "resolved absolute soure map url", "url", sourceURL, "sourceMapURL", resolvedSourceMapURL)
+	}
+	level.Debug(store.l).Log("msg", "attempting to download sourcemap file", "url", resolvedSourceMapURL)
+	result, err = store.downloadFileContents(resolvedSourceMapURL)
+	if err != nil {
+		level.Debug(store.l).Log("failed to download source map file", "url", resolvedSourceMapURL, "err", err)
+		return nil, "", err
+	}
+	return result, resolvedSourceMapURL, nil
+}
+
+func (store *RealSourceMapStore) getSourceMapFromFileSystem(sourceURL string, release string, fileconf config.SourceMapFileLocation) (content []byte, sourceMapURL string, err error) {
+	if len(sourceURL) == 0 || !strings.HasPrefix(sourceURL, fileconf.MinifiedPathPrefix) || strings.HasSuffix(sourceURL, "/") {
+		return nil, "", nil
+	}
+	pathParts := []string{strings.Replace(fileconf.Path, "{RELEASE}", cleanForFilePath(release), 1)}
```

Comment:
nit/suggestion: Instead of using our custom templating language, we use go's [template](https://pkg.go.dev/text/template) package (i.e. "{{ .Release }}" or whatnot)

Reply:
good point!
- [ ] use go templating lang